model/src/main/java/oracle/kubernetes/weblogic/domain/model/Domain.java

```diff
@@ -460,4 +474,57 @@ public boolean equals(Object other) {
         .append(status, rhs.status)
         .isEquals();
   }
+
+  public List<String> getValidationFailures() {
+    return new Validator().getValidationFailures();
+  }
+
+  class Validator {
+    private List<String> failures = new ArrayList<>();
+    private Set<String> clusterNames = new HashSet<>();
+    private Set<String> serverNames = new HashSet<>();
+
+    List<String> getValidationFailures() {
+      addDuplicateNames();
+      addUnmappedLogHome();
+
+      return failures;
+    }
+
+    private void addDuplicateNames() {
+      getSpec().getManagedServers().stream().map(ManagedServer::getServerName).forEach(this::checkDuplicateServerName);
+      getSpec().getClusters().stream().map(Cluster::getClusterName).forEach(this::checkDuplicateClusterName);
+    }
+
+    private void checkDuplicateServerName(String s) {
+      if (serverNames.contains(s))
+        failures.add(String.format("More than one server is named '%s'", s));
```

Comment:
How could this ever happen unless there was a bug in our topology generator?   
Regardless 'server' --> 'WebLogic Server in your WebLogic domain'

Reply:
Note that it _is_ possible for two servers to have the same name if they both happen to resolve to same DNS1123 form (all lower cases, underscores converted to hyphens).  There's a pending JIRA to check this in the introspector, but maybe it'd fine to check it here instead.

Reply:
This is checking the domain resource, not the WLS domain.

Reply:
ok

Reply:
Actually, not 100% OK.  It'd be helpful if the error message clearly stated that this is a domain resource issue, and maybe which specific part of the schema was messed up.

Reply:
When you say that it is possible, do you mean that it is allowed? Or that we have to reject it?

That is, is it legal to define one server as "My_Server" and another as "my-server"? Doing so would cause a problem with pod and service names.

Reply:
Since this particular error message is about the user-supplied domain resource, that should be made clear.

As for My_Server/my-server that's a separate issue.  If operator isn't in the picture, it is legal to define My_Server and my-server.  If the operator is in the picture, this is not legal due to the service name issue as you point out, but we don't have any checks for this currently.  We could do the checks in the introspector but maybe it'd be better to do them in the Operator? (full details in pending JIRA OWLS-74954, which assumes the introspector should have the checks).